server.js

```diff
@@ -29,8 +29,19 @@ app.get("/weather-object", (req, res) => {
 });
 
 app.post("/weather-object", (req, res) => {
-  projectData.temperature = req.body.temperature;
-  projectData.date = req.body.date;
-  projectData.userResponse = req.body.userResponse;
-  res.send(projectData);
+  try {
+    
+    const {temperature, date, userResponse} = req.body;
+    if (!temperature || !date || !userResponse) {
+      return res.status(400).send({ status: 400, message: "temperature, date & userResponse are required!" });
+    }
```

Comment:
I changed a little the code but does the same, I am conducting a basic validation here, in case an error happens, we need to **return** ** the response to end the server routine and return properly the errors.